inmoose/cohort_qc/cohort_metric.py

```diff
@@ -0,0 +1,864 @@
+            sns.violinplot(
+                ax=axs[0],
+                x="Covariate_Combination",
+                y="Mean_Expression",
+                data=mean_before,
+            )
+            axs[0].set_title(
+                "Mean of sample gene expression distribution before correction"
+            )
+            axs[0].set_xlabel("Covariate combination")
+            axs[0].set_ylabel("Mean gene expression by sample")
+            axs[0].tick_params(axis="x", rotation=45)
+            ax = axs[1]
+        else:
+            fig, ax = plt.subplots(nrows=1, ncols=1, figsize=(width, heigth / 2))
+
+        # Plot after correction
+        sns.violinplot(
+            ax=ax,
+            x="Covariate_Combination",
+            y="Mean_Expression",
+            data=mean_after,
+        )
+        ax.set_title("Mean of sample gene expression distribution after correction")
+        ax.set_xlabel("Covariate combination")
+        ax.set_ylabel("Mean gene expression by sample")
+        ax.tick_params(axis="x", rotation=45)
+
+        # Return the plot as a figure object
+        fig.tight_layout()
+        return fig
+
+    def silhouette_score(self) -> tuple:
+        """
+        Compute the Silhouette Score for batch mixing before and after batch correction.
+
+        Returns
+        -------
+        tuple
+            Silhouette scores before and after correction.
+        """
+        # Compute silhouette score for after batch correction
+        score_after = silhouette_score(
+            self.data_expression_df.T, self.clinical_df[self.batch_column]
+        )
+
+        if self.data_expression_df_before is not None:
+            # Compute silhouette score before correction if available
+            score_before = silhouette_score(
+                self.data_expression_df_before.T, self.clinical_df[self.batch_column]
+            )
+        else:
+            score_before = None
+
+        return score_before, score_after
+
+    def compute_entropy(
+        self, data_expression_df: pd.DataFrame, n_neighbors: int
+    ) -> float:
+        """
+        Compute the entropy of batch mixing to evaluate the effectiveness of batch correction.
+
+        Parameters
+        ----------
+        data_expression_df : pandas.DataFrame
+            Gene expression data frame to evaluate.
+        n_neighbors : int
+            Number of nearest neighbors to consider in the entropy calculation.
+
+        Returns
+        -------
+        float
+            Mean entropy value indicating the level of batch mixing.
+        """
+
+        nbrs = NearestNeighbors(n_neighbors=n_neighbors, metric="euclidean").fit(
+            data_expression_df.T
+        )
+        _, indices = nbrs.kneighbors(data_expression_df.T)
+        batch_labels = self.clinical_df[self.batch_column].values
+
+        entropies = []
+        for idx in range(indices.shape[0]):
+            neighbors_labels = batch_labels[indices[idx]]
```

Comment:
Here a `for indice in indices` is enough

Reply:
[picky mode]
the singular of "indices" is "index"
[/picky mode]